src/modules/matches/matches.service.ts

```diff
@@ -2,30 +2,61 @@ import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository, UpdateResult, DeleteResult } from 'typeorm';
 import { v4 as uuidv4 } from 'uuid';
+import { Team } from '../teams/team.entity';
+import { CreateMatchDto, MatchDto, UpdateMatchDto } from './dto';
 import { Match } from './match.entity';
 
 @Injectable()
 export class MatchesService {
   constructor(
     @InjectRepository(Match)
-    private matchesRepository: Repository<Match>
+    private matchesRepository: Repository<Match>,
+    @InjectRepository(Team)
+    private readonly teamsRepository: Repository<Team>
   ) {}
 
-  async findAll(): Promise<Match[]> {
-    return this.matchesRepository.find();
+  async findAll(): Promise<MatchDto[]> {
+    const matches = await this.matchesRepository.find({
+      relations: { home_team: true, away_team: true }
+    });
+
+    return await Promise.all(
+      matches.map(async (match) => {
+        return await MatchDto.fromEntity(match);
+      })
+    );
   }
 
-  async findOneById(id: string): Promise<any> {
-    return this.matchesRepository.findOneOrFail({ where: { id: id } });
+  async findOneById(id: string): Promise<MatchDto> {
+    const match = await this.matchesRepository.findOneOrFail({
+      where: { id: id },
+      relations: { home_team: true, away_team: true }
+    });
+    return await MatchDto.fromEntity(match);
   }
 
-  async create(match: Match): Promise<Match> {
+  async create(matchData: CreateMatchDto): Promise<MatchDto> {
+    const match = new Match();
     match.id = uuidv4();
-    return this.matchesRepository.save(match);
+
+    match.home_team = await this.teamsRepository.findOneOrFail({ where: { id: matchData.home_team } });
+    match.home_score = 0;
+
+    match.away_team = await this.teamsRepository.findOneOrFail({ where: { id: matchData.away_team } });
+    match.away_score = 0;
+
+    const savedMatch = await this.matchesRepository.save(match);
+
+    return await MatchDto.fromEntity(savedMatch);
   }
 
-  async update(id: string, match: Match): Promise<UpdateResult> {
+  async update(id: string, matchData: UpdateMatchDto): Promise<UpdateResult> {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏i was able to update a match with negative value 😉

Reply:
well spotted, I'll do a dedicated MR to implement data validation